stdlib/public/core/FloatingPoint.swift

```diff
@@ -328,15 +328,24 @@ public protocol FloatingPoint: SignedNumeric, Strideable, Hashable
   /// `infinity` is greater than this value.
   static var greatestFiniteMagnitude: Self { get }
 
-  /// The mathematical constant pi.
+  /// The [mathematical constant π][wiki], approximately equal to 3.14159.
   ///
-  /// This value should be rounded toward zero to keep user computations with
-  /// angles from inadvertently ending up in the wrong quadrant. A type that
-  /// conforms to the `FloatingPoint` protocol provides the value for `pi` at
-  /// its best possible precision.
+  /// This constant is most often defined as the numerical value of the ratio
+  /// of a circle's circumference to its diameter, which is the same for all
+  /// circles.
+  ///
+  /// A common application of this constant in geometry is to measure a half
+  /// revolution in radians. Used like this, it is equivalent to 180 degrees.
```

Comment:
I wonder if we can condense this into the preceding paragraph.

I do agree that the introductory paragraph is an improvement. As to the other additions, we ought to have a good sense of who the _target audience_ is, bearing in mind that this documentation appears in many contexts (for example, in a relatively small panel in Xcode).

Consider a busy user who happens to click on `.pi` in their code: they're much more likely to find it helpful to know how the value is rounded (to keep angles from inadvertently ending up in the wrong quadrant) than to need a reminder that the ratio of a circle's circumference to its diameter is the same for all circles. As proposed in this PR, the likely more salient information is pushed further down and potentially hidden if the user doesn't scroll.

A sentence or two so that a user can see at a glance that `.pi` is indeed the constant they're thinking of is great; more than that, I'd argue, is beyond the role of the Swift standard library documentation to teach.

Reply:
I think that's fair. I considered pushing the definition down to promote the more practical information, but it would break textual cohesion, as the application in geometry is an extension of its geometric definition. I guess it makes sense to omit the definition altogether for the sake of conciseness, though. To be honest I was hoping that Swift would adopt τ some day, in which case having the definition would help highlight the distinction between the two constants. But oh well, I guess we can cross that bridge if we ever get to it.